src/components/Accordion.tsx

```diff
@@ -76,51 +79,62 @@ export function Accordion<X extends Only<AccordionXss, X>>(props: AccordionProps
   }, [expanded, setContentHeight]);
   useResizeObserver({ ref: contentRef, onResize });
 
+  const toggle = useCallback(() => {
+    setExpanded((prev) => !prev);
+    if (setExpandedIndex) setExpandedIndex(index);
+  }, [index, setExpandedIndex]);
+
+  const touchableStyle = useMemo(
+    () => ({
+      ...Css.df.jcsb.gapPx(12).aic.p2.baseMd.outline("none").onHover.bgGray100.if(!!titleOnClick).baseSb.$,
+      ...(compact && Css.smMd.pl2.prPx(10).py1.bgGray100.mbPx(4).br8.onHover.bgGray200.$),
+      ...(compact && !!titleOnClick && Css.br0.$),
+      ...(disabled && Css.gray500.$),
+      ...(isFocusVisible && Css.boxShadow(`inset 0 0 0 2px ${Palette.Blue700}`).$),
+      ...xss,
+    }),
+    [compact, disabled, isFocusVisible, titleOnClick, xss],
+  );
+
   return (
     <div
-      {...testIds.container}
+      {...tid.container}
       css={{
         ...Css.bGray300.if(topBorder).bt.if(bottomBorder).bb.$,
         ...(size ? Css.wPx(accordionSizes[size]).$ : {}),
       }}
     >
-      <button
-        {...testIds.title}
-        {...focusProps}
-        aria-controls={id}
-        aria-expanded={expanded}
-        disabled={disabled}
-        css={{
-          ...Css.df.jcsb.gapPx(12).aic.w100.p2.baseMd.outline("none").onHover.bgGray100.$,
-          ...(compact && Css.smMd.pl2.prPx(10).py1.bgGray100.mbPx(4).br8.onHover.bgGray200.$),
-          ...(disabled && Css.gray500.$),
-          ...(isFocusVisible && Css.boxShadow(`inset 0 0 0 2px ${Palette.Blue700}`).$),
-          ...xss,
-        }}
-        onClick={() => {
-          setExpanded(!expanded);
-          if (setExpandedIndex) setExpandedIndex(index);
-        }}
-      >
-        <span css={Css.fg1.tl.$}>{title}</span>
-        <span
-          css={{
-            ...Css.fs0.$,
-            transition: "transform 250ms linear",
-            transform: expanded ? "rotate(180deg)" : "rotate(0deg)",
-          }}
+      {titleOnClick ? (
+        <div {...focusProps} aria-controls={id} aria-expanded={expanded} css={Css.df.jcsb.$}>
+          <button {...tid.title} disabled={disabled} css={{ ...touchableStyle, ...Css.fg0.$ }} onClick={titleOnClick}>
+            {title}
+          </button>
+          <button {...tid.toggle} disabled={disabled} css={{ ...touchableStyle, ...Css.fg1.jcfe.$ }} onClick={toggle}>
```

Comment:
I think personally I would switch the fg0/1 and have the chevon CTA be smaller, but :shrug: could really see that going either way/defer to you/design.